### PR TITLE
thanos-sidecar support external Thanos clusters

### PR DESCRIPTION
**Pull request by nicholasgibson2**
By default thanos-sidecar sets the cluster and grpc addresses to the internal k8s address. In order to support communication from external clusters you have to set cluster.advertise-address and grpc-advertise-address to use the external k8s host address.

### CODE REVIEW

**Comment by metalmatze** [comment]
lgtm

Can you please run `make clean generate` from the root of the repository once more? Thanks

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -326,6 +326,12 @@ type ThanosSpec struct {
 	S3 *ThanosS3Spec `json:"s3,omitempty"`
 	// ObjectStorageConfig configures object storage in Thanos.
 	ObjectStorageConfig *v1.SecretKeySelector `json:"objectStorageConfig,omitempty"`
+	// Explicit (external) host:port address to advertise for gRPC StoreAPI in gossip cluster.
+	// If empty, 'grpc-address' will be used..
```
Comment (brancz) [inline pkg/apis/monitoring/v1/types.go]: Trailing `.`

**Comment by nicholasgibson2** [comment]
@metalmatze @brancz Alright think I got the docs fixed

**Comment by metalmatze** [comment]
Thanks! 
I'm going to merge even though GitHub says Travis is pending, but Travis itself is green. :man_shrugging:

**Pull request merged**